planning/planning_scene_monitor/src/planning_scene_monitor.cpp

```diff
@@ -627,6 +627,8 @@ void planning_scene_monitor::PlanningSceneMonitor::excludeRobotLinksFromOctree()
 
   includeRobotLinksInOctree();
   const std::vector<const robot_model::LinkModel*> &links = getRobotModel()->getLinkModelsWithCollisionGeometry();
+  ros::WallTime start = ros::WallTime::now();
+  bool warned = false;
```

Comment:
Didn't even know that existed.  If we do this, we must keep evaluating
(ros::WallTime::now() - start) > ros::WallDuration(30.0)) on every link (even after we have warned).  But it's not very costly as has been pointed out.